lib/puppet/ssl/configuration.rb

```diff
@@ -50,7 +50,10 @@ def decode_cert_bundle(bundle_str)
 
   # read_file makes testing easier.
   def read_file(path)
-    File.read(path)
+    # https://www.ietf.org/rfc/rfc2459.txt defines the x509 V3 certificate format
+    # CA bundles are concatenated X509 certificates, but may also include
+    # comments, which could have UTF-8 characters
```

Comment:
👍  - in earlier work on this I was indeed working with certs with UTF-8 comments. Had to work a bit to make it happen, and it was outside puppet's cert generation, but it worked.